packages/health-and-info/src/index.ts

```diff
@@ -0,0 +1,195 @@
+import AgentOptions, { HttpOptions, HttpsAgent, HttpsOptions } from 'agentkeepalive'
+import superagent from 'superagent'
+import { Request, Response, NextFunction } from 'express'
+import {
+  ComponentCheckResult,
+  ExternalComponent,
+  CustomComponent,
+  ApplicationInfo,
+  HealthCheckOptions,
+  HealthCheckResult,
+} from './types/HealthCheckOptions'
+
+export default class HealthCheck {
+  private defaultOptions = {
+    components: {
+      retries: 2,
+      timeout: {
+        response: 1500,
+        deadline: 2000,
+      },
+      enabled: true,
+    },
+  }
+
+  constructor(
+    private readonly options: HealthCheckOptions,
+    private readonly logger: Console = console,
+  ) {}
+
+  private getDeploymentInfo = (appInfo: ApplicationInfo) => {
+    return {
+      build: {
+        buildNumber: appInfo.buildNumber,
+        gitRef: appInfo.gitRef,
+      },
+      version: appInfo.buildNumber,
+      uptime: Math.floor(process.uptime()),
+    }
+  }
+
+  private checkExternalComponents = (components: Partial<ExternalComponent | CustomComponent>[]) => {
+    return components
+      .filter(component => !('healthCheckFunction' in component))
+      .map(component => ({ ...this.defaultOptions.components, ...component }) as ExternalComponent)
+      .filter(component => component.enabled)
+      .map(this.checkExternalComponent)
+  }
+
+  private checkExternalComponent = async (component: ExternalComponent): Promise<ComponentCheckResult> => {
+    const { timeout, retries } = component
+    const agent = component.url.startsWith('https')
+      ? new HttpsAgent(component.agentConfig as HttpsOptions)
+      : new AgentOptions(component.agentConfig as HttpOptions)
```

Comment:
**issue:** These [Agents](https://nodejs.org/api/http.html#class-httpagent) are heavy weight and designed to be long lived. Worth creating one per api rather than on every request

Reply:
I think, if I'm following your suggestion in https://github.com/ministryofjustice/hmpps-typescript-lib/pull/7#discussion_r1762501992, we should be able to handle this as part of that.